api/server/docker.go

```diff
@@ -169,79 +215,132 @@ func (d *driver) mountpath(name string) string {
 	return path.Join(volume.MountBase, name)
 }
 
+func (d *driver) getConn() (*grpc.ClientConn, error) {
+	if d.conn == nil {
```

Comment:
i hadn't placed any locks here, but we should have locks on this function so that we do not connect many times before d.conn becomes non-null

Reply:
Good idea, will add